src/main/java/com/ocient/jdbc/XGConnection.java

```diff
@@ -3976,32 +3975,39 @@ public void reconnect(final boolean shouldRequestVersion) throws IOException, SQ
 		}
 	}
 
-	private void clientHandshakeFinalize() throws SQLException, Exception {
+	// FIXME
+	// Blocking IO
+	private CompletionStage<Void> clientHandshakeFinalize() throws SQLException, Exception {
 		// Set the default schema on the connection if one has not already been set
 		resetSchema().ifPresent(schema -> {
 			try
 			{
+				// Blocking IO
+				// Connection superclass requires setSchema to return void
+				// so CompletionStage<Void> cannot be used here
 				setSchema(schema);
 			}
 			catch (final Exception e)
 			{
 			}
 		});
 
+		CompletionStage<Void> setPsoCompletionStage = null;
```

Comment:
(nit)

I typically avoid nulls if possible, and if something can legitimately be nullable, I'll leave an indicator in the variable name. In this case, the asynchronous execution does not produce a result so what you really want a way to say there is nothing to do.
```suggestion
		CompletionStage<Void> setPsoCompletionStage = StageSupport.voidStage();
```
This way, you can get rid of the null check and always return `setPsoCompletionStage`.

Just a little defensive programming trick.